### PR TITLE
Conform to testify's types.

### PR DESCRIPTION
**Pull request by elisarver**
Previously the functions `AssertError` and `ValidateError` took their message as the first term. Testify's Assert and Require use a value called `msgAndArgs` to perform `printf`-like pass-through of their arguments for the failure case. This change corrects `AssertError` and  `ValidateError` to conform to the format provided by testify's libraries.

This change also uses testify's version of `*testing.T` in interface form to enforce the contract between the testify libraries on upgrade.

### CODE REVIEW

**Comment by elisarver** [comment]
I realize I was half-in and half-out of implementing part of this, and the extra "testingt" package was super unnecessary, so I'm pulling it out and using counterfeiter to generate the mocks to said interface at their point of use.

**Comment by elisarver** [comment]
Much better. Got docs in place, and also removed the redundant testingt.TestingT type and using either assert or require's version for each case.

File: assert/assert.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import (
+	"github.com/stretchr/testify/assert"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 github.com/stretchr/testify/assert.TestingT
+
+// Assertions adds assertions to testify's assert lib.
+type Assertions struct {
+	*assert.Assertions
```
Comment (bkyoung) [inline assert/assert.go:10]: I think it is best to name this field explicitly, even though it is legal to not do so, and it has a name implicitly.  It avoids the confusion below in which you explicitly provide a name to this field, and order the fields of this struct differently.

File: assert/assert.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+		tt:         tt,
+		Assertions: assert.New(tt),
+	}
+}
+
+// WantError asserts that a function returned an error (i.e. not `nil`).
+//
+//   actualObj, err := SomeFunction()
+//   success := a.WantError(test.wantErr, err)
+func (a *Assertions) WantError(wantErr bool, err error, msgAndArgs ...interface{}) (success bool) {
```
Comment (bkyoung) [inline assert/assert.go:26]: I _love_ this pattern.  It is so simple and elegant.

File: require/require.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+import (
+	"github.com/stretchr/testify/require"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 github.com/stretchr/testify/require.TestingT
+
+// Assertions adds assertions to testify's require lib.
+type Assertions struct {
+	*require.Assertions
```
Comment (bkyoung) [inline require/require.go:10]: Same as the assert package, it is probably best to explicitly name this field

**Review by bkyoung — APPROVED** [review]
I am a big fan of the value add these extensions provide to existing tooling/libraries.  This is such an elegant use of interfaces and helper functions.  I will be looking for opportunities to employ this because it is a great improvement for a small amount of additional code.  Technical leverage FTW.

File: assert/assert.go (hunk 1 above)
Comment (elisarver) [reply assert/assert.go:10]: I'll add an explicit comment to state that this is done for embedding purposes. If I name it in the type definition, I lose the embedding.

File: require/require.go (hunk 3 above)
Comment (elisarver) [reply require/require.go:10]: same answer as above.

**Comment by elisarver** [comment]
> I will admit I do not feel comfortable in my ability to fully grok the test suite/mocks/etc to assess this entire changeset without an up-front time investment to go through and poke around.

I have added some description of why I import like I do, why we use a mock instead of a real testing.T (it would fail the real test) and so on. Give it a once-over, and see if clarity results; if not, let me know.

**Pull request merged**